### ISSUE TITLE
Baggage namespace has conflict with `CoreBaggage.Baggage`

### ISSUE THREAD

**Issue by fabianfett** [apple/swift-distributed-tracing-baggage#11]
This package produces a library called `Baggage`.

The library `Baggage` uses `@_exported import CoreBaggage`. 
`CoreBaggage` has a `struct Baggage`.

For this reason it is not possible to use the `Baggage` namespace. The compiler will always detect `Baggage` as the struct form `CoreBaggage`.

Failing code:

```swift
import Baggage

extension Lambda.Context: Baggage.LoggingContext {}
```

Produces the compile error: `'LoggingContext' is not a member of struct 'CoreBaggage.Baggage'.`

**Comment by ktoso** [comment]
Paging @tomerd @weissi @Lukasa to discuss this one...

Originally when we prepared the packages we intentionally made the packages `CoreBaggage` which contains `Baggage` and `Baggage` which contains `LoggingContext`. It was deemed fine at the time, but seems we have an issue.

The baggage type will remain even with task local values present I think, although as usual it is "very hidden" and noone should need to touch it except a few tracer implementations.

So this is more about the `LoggingContext` being a name which may need disambiguation... though I'd argue others should not define their own `LoggingContext` and could call them `MyFrameworkLoggingContext`, which again solves the issue then.

I think we're also working to improve the namespacing issues if I remember correctly @tomerd ?

**Comment by ktoso** [comment]
We could do `BaggageModule` I guess similar to what stdlib team projects do https://github.com/apple/swift-collections/tree/main/Sources

**Comment by Lukasa** [comment]
+1 for `BaggageModule`.

**Comment by slashmo** [comment]
+1 on `BaggageModule` too.

**Comment by ktoso** [comment]
Sounds good -- could you send in a PR @fabianfett ? I'll tag a release then and update the dependency in the main tracing repo so noone should really notice the change in practice.

Thanks for noticing!

**Comment by slashmo** [comment]
Closed via #13, the module name is now `InstrumentationBaggage`.
